include/seqan3/search/algorithm/detail/search_trivial.hpp

```diff
@@ -63,16 +74,21 @@ inline bool search_trivial(cursor_t cur, query_t & query, typename cursor_t::siz
     // Approximate case
     else
     {
+        // Do not use insertion if there is a match.
+        // Check if cursor did a step in the index.
+        bool const do_insertion = cur.query_length() > 0 ? cur.last_rank() != to_rank(query[query_pos]) : true;
+
         // Insertion
-        if (error_left.insertion > 0)
+        // Do not allow insertions after an deletion.
+        if (do_insertion && ((memory != last_error::DELETION) || error_left.substitution == 0) && error_left.insertion > 0)
```

Comment:
```suggestion
        if (do_insertion && (memory != last_error::DELETION || error_left.substitution == 0) && error_left.insertion > 0)
```

I think you don't need these parenthesis.
Same for the other new if clauses

Reply:
I tried it for the other if clauses, but the compiler to seems to be configure to complain about that:
error: suggest parentheses around ‘&&’ within ‘||’ [-Werror=parentheses]